lib/std/sysrand.nim

```diff
@@ -0,0 +1,287 @@
+        inc(result, readBytes)
+      else:
+        if osLastError().int in {EINTR, EAGAIN}:
+          discard
+        else:
+          result = -1
+          break
+
+elif defined(openbsd):
+  proc getentropy(p: pointer, size: cint): cint {.importc: "getentropy", header: "<unistd.h>".}
+    # fills a buffer with high-quality entropy,
+    # which can be used as input for process-context pseudorandom generators like `arc4random`.
+    # The maximum buffer size permitted is 256 bytes.
+
+  proc getRandomImpl(p: pointer, size: int): int {.inline.} =
+    result = getentropy(p, cint(size)).int
+
+  template urandomImpl(dest: var openArray[byte]) =
+    batchImpl(result, dest, getRandomImpl)
+
+elif defined(freebsd):
+  type cssize_t {.importc: "ssize_t", header: "<sys/types.h>".} = int
+
+  proc getrandom(p: pointer, size: csize_t, flags: cuint): cssize_t {.importc: "getrandom", header: "<sys/random.h>".}
+    # Upon successful completion, the number of bytes which were	actually read
+    # is returned. For requests larger than 256	bytes, this can	be fewer bytes
+    # than were requested. Otherwise, -1 is returned and the global variable
+    # errno is set to indicate the error.
+
+  proc getRandomImpl(p: pointer, size: int): int {.inline.} =
+    result = getrandom(p, csize_t(batchSize), 0)
+
+  template urandomImpl(dest: var openArray[byte]) =
+    batchImpl(result, dest, getRandomImpl)
+
+elif defined(ios):
+  {.passL: "-framework Security".}
+
+  const errSecSuccess = 0 ## No error.
+
+  type
+    SecRandom {.importc: "struct __SecRandom".} = object
+
+    SecRandomRef = ptr SecRandom
+      ## An abstract Core Foundation-type object containing information about a random number generator.
+
+  proc secRandomCopyBytes(
+    rnd: SecRandomRef, count: csize_t, bytes: pointer
+    ): cint {.importc: "SecRandomCopyBytes", header: "<Security/SecRandom.h>".}
+    ## Generates an array of cryptographically secure random bytes.
+
+  template urandomImpl(dest: var openArray[byte]) =
+    let size = dest.len
+    if size == 0:
+      return
+
+    # `kSecRandomDefault` is a synonym for NULL.
+    result = secRandomCopyBytes(nil, csize_t(size), addr dest[0])
+
+elif defined(macosx):
+  const sysrandomHeader = """#include <Availability.h>
+#include <sys/random.h>
+"""
+
+  proc getentropy(p: pointer, size: csize_t): cint {.importc: "getentropy", header: sysrandomHeader.}
+    # getentropy() fills a buffer with random data, which can be used as input 
+    # for process-context pseudorandom generators like arc4random(3).
+    # The maximum buffer size permitted is 256 bytes.
+
+  proc getRandomImpl(p: pointer, size: int): int {.inline.} =
+    result = getentropy(p, csize_t(size)).int
+
+  template urandomImpl(dest: var openArray[byte]) =
+    batchImpl(result, dest, getRandomImpl)
+else:
+  template urandomImpl(dest: var openArray[byte]) =
+    let size = dest.len
+    if size == 0:
+      return
+
+    let fd = posix.open("/dev/urandom", O_RDONLY)
```

Comment:
```suggestion
    let fd = posix.open("/dev/urandom", O_RDONLY)
    # see: https://www.2uo.de/myths-about-urandom/ which justifies using urandom instead of random
```